android/demos/compose/src/main/kotlin/com/uber/rib/compose/root/main/logged_in/ScoreStream.kt

```diff
@@ -15,27 +15,30 @@
  */
 package com.uber.rib.compose.root.main.logged_in
 
-import com.jakewharton.rxrelay2.BehaviorRelay
-import io.reactivex.Observable
+import com.uber.rib.core.RibDispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.withContext
 
 class ScoreStream(playerOne: String, playerTwo: String) {
 
-  private val scoresRelay: BehaviorRelay<Map<String, Int>> = BehaviorRelay.createDefault(
+  private val scoresFlow = MutableStateFlow(
     mapOf(
       playerOne to 0,
       playerTwo to 0
     )
   )
 
-  fun addVictory(userName: String) {
-    val scores = (scoresRelay.value ?: emptyMap()).toMutableMap()
+  suspend fun addVictory(userName: String) = withContext(RibDispatchers.IO) {
```

Comment:
I think `addVictory` should update the flow atomically, as there's potential to have concurrent coroutines calling `addVictory`.

```kotlin
  suspend fun addVictory(userName: String) = withContext(RibDispatchers.IO) {
    scoresFlow.update { scores ->
      scores.toMutableMap().apply {
        if (userName in scores) {
          set(userName, getValue(userName) + 1)
        }
      }.toMap()
    }
  }
```

Reply:
That's a reasonable suggestion